Weather/Modules/CityDetail/Interface/CityDetailInterfaces.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+import UIKit
+
+
+/// The idea behind keeping all protocol declarations for all VIPER module building blocks
+/// (Router, Builder, Presenter, Interactor, ViewModel and View) in one file
+/// is to make it more explicit that they work together to construct a VIPER module.
+
+
+// ------------------------------- //
+// -- Builder:
+// ------------------------------- //
+
+protocol WeatherDetailBuilder {
+    
+    func buildWeatherDetailModule(city: String) -> UIViewController?
+}
+
+
+// ------------------------------- //
+// -- Interactor:
+// ------------------------------- //
+
+enum FetchCityWeatherResult {
+    
+    case Success(weather: WeatherData)
+    case Failure(reason: NSError)
+}
+
+protocol WeatherDetailInteractor {
+    
+    func fetchCityWeather(city: String, completion: (FetchCityWeatherResult) -> ())
+}
+
+
+// ------------------------------- //
+// -- View Models:
+// ------------------------------- //
+
+struct WeatherDetailViewModel {
+    
+    let cityName    : String
+    let temperature : String
+    let forecasts   : [WeatherDetailForecastViewModel]
+}
+
+struct WeatherDetailForecastViewModel {
+    let day : String
+    let temp: String
+}
+
+
+// ------------------------------- //
+// -- Presenter:
+// ------------------------------- //
+
+protocol WeatherDetailPresenter: class {
+    
+    var         city        : String                    {get}
+    
+    /// Required dependencies for that presenter in VIPER architecture.
+    /// The idea is to make it explicit in protocol what other components
+    /// are needed for this module presenter to function properly.
+    // ???: Is that a good idea to declare those dependencies in protocol,
+    // ???: or should they be a part of contrete implementation instead?
+    
+    var         interactor  : WeatherDetailInteractor   {get set}
```

Comment:
Good question! 
The idea of a protocol is of a "contract" of usage. So exposing variables here would somehow give away "some" implementation details and also allow consumers to "use" directly the interactor or the view, which is something we don't want to happen ideally.